zfit/minimizers/fitresult.py

```diff
@@ -40,39 +43,69 @@ def _hesse_np(result: "FitResult", params, sigma=1.0):
     if sigma != 1.0:
         raise ValueError("sigma other then 1 is not valid for hesse numpy.")
 
-    # check if no weights in data
-    if any([data.weights is not None for data in result.loss.data]):
-        raise WeightsNotImplementedError("Weights are not supported with hesse numpy.")
-
-    # minimizer = fitresult.minimizer
-    # from zfit.minimizers.minimizer_minuit import Minuit
-    # if not isinstance(minimizer, Minuit):
-    #     raise TypeError("Cannot perform hesse error calculation 'minuit' with a different minimizer then"
-    #                     "`Minuit`.")
-    result_hesse = 2 * np.linalg.inv(result.loss.value_gradients_hessian(params)[2])
+    covariance = _covariance_np(result, params)
```

Comment:
What about the weights? Any good idea?
Guess RooFit has a "square the weights, recalculate the likelihood" approximation. Anything better in mind by chance?

Reply:
Can the hessian matrix can be computed using the weights?

Reply:
The procedure is actually described in `Statistical Methods in Experimental Physics` chapter 8.5.2, page 198 if you have it at hand.
![Scan_marinang_110320200990_001 - copie](https://user-images.githubusercontent.com/24250309/76403330-0aba4080-6385-11ea-8a13-5acd35cd5125.jpg)

Reply:
So as you said squaring the weights and recompute the likelihood.

Reply:
But I think you don't have to refit, just clone the loss and square the weights compute the H prime matrix. In that case I think a caching of the covariance matrix should be done in order to avoid the repeat the loss cloning step.

Reply:
Here https://arxiv.org/pdf/1911.01303.pdf an asymptotic method is described to compute the covariance matrix.

Reply:
Let's move that discussion to another PR, one thing at a time